### ISSUE TITLE
state_history_plugin: incorrect serialization of WA keys

### ISSUE THREAD

**Issue by cc32d9** [EOSIO/eos#9087]
as of release 2.0.5:

ship is using `eosio::chain::shared_public_key` for serializing key weight:

```
template <typename ST>
datastream<ST>& operator<<(datastream<ST>& ds, const history_serial_wrapper<eosio::chain::shared_key_weight>& obj) {
   fc::raw::pack(ds, as_type<eosio::chain::shared_public_key>(obj.obj.key));
   fc::raw::pack(ds, as_type<uint16_t>(obj.obj.weight));
   return ds;
}
```

But `shared_public_key` is defining its data as 
```
using shared_public_key_data = fc::static_variant<fc::ecc::public_key_shim, fc::crypto::r1::public_key_shim, shared_string>;
```

so, in case of WA key, the FC serializer is packing it as a string. Then abieos is unable to deserialize it because it expects a variant.

The WA signatures seem to be safe, so it's only the authority objects affected.

**Comment by cc32d9** [comment]
not a variant, abieos is expecting user presence byte, but it gets the string length. nevertheless, it's wrongly serialized.

**Comment by spoonincode** [comment]
Fix merged for 2.0.6 release thanks for report
